### PR TITLE
airflow example

### PR DESCRIPTION
**Pull request by kshitij12345**
# Description

Include a summary of the changes and the related issue.

__Related to:__ <ClickUp/JIRA task name>

__Any expected test failures?__

---

Add a `[X]` to relevant checklist items

## ❔ This change

- [x] adds a new feature
- [ ] fixes breaking code
- [ ] is cosmetic (refactoring/reformatting)

---

## ✔️ Pre-merge checklist

- [ ] Refactored code ([sourcery](https://sourcery.ai/))
- [x] Tested code locally
- [x] Precommit installed and run before pushing changes
- [ ] Added code to GitHub tests ([notebooks](workflows/test-notebooks.yml), [scripts](workflows/test-scripts.yml))
- [x] Updated GitHub [README](../README.md)
- [ ] Updated the projects overview page on Notion

---

## 🧪 Test Configuration

- OS: Ubuntu
- Python version: 3.10
- Neptune version: 1.4.1
- Affected libraries with version: Airflow 2.6.3 , Tensorflow 2.13.0

### CODE REVIEW

File: integrations-and-supported-tools/airflow/scripts/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+neptune-airflow @ git+https://github.com/neptune-ai/neptune-airflow@main
```
Comment (kshitij12345) [inline integrations-and-supported-tools/airflow/scripts/requirements.txt:1]: To be updated once released.

File: integrations-and-supported-tools/airflow/scripts/run_examples.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+set -e
+
+echo "Installing Airflow with constraints..."
+pip install "apache-airflow==2.6.3" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.6.3/constraints-3.7.txt"
```
Comment (kshitij12345) [inline integrations-and-supported-tools/airflow/scripts/run_examples.sh:4]: It is recommended to have constraints file: https://airflow.apache.org/docs/apache-airflow/stable/installation/installing-from-pypi.html#why-we-need-constraints

**Comment by kshitij12345** [comment]
@SiddhantSadangi Ping for preliminary review. Also, we should we log it to a separate project?

**Comment by SiddhantSadangi** [comment]
> we should we log it to a separate project?
yup. I'll take care of that

File: integrations-and-supported-tools/airflow/scripts/Neptune_Airflow.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            optimizer=optimizer, loss="sparse_categorical_crossentropy", metrics=["accuracy"]
+        )
+
+        # (Neptune) log metrics during training using the above handler
+        neptune_cbk = NeptuneCallback(run=handler)
+        model.fit(x_train, y_train, epochs=5, batch_size=64, callbacks=[neptune_cbk])
+
+        model.save("my_model.h5")
+
+        # If task don't run on same machine then upload checkpoint.
```
Comment (normandy7) [inline integrations-and-supported-tools/airflow/scripts/Neptune_Airflow.py:55]:
```suggestion
        # If task doesn't run on same machine, then upload checkpoint
```

File: integrations-and-supported-tools/airflow/scripts/Neptune_Airflow.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    schedule="@daily",
+    start_date=datetime.today() - timedelta(days=1),
+    catchup=False,
+) as dag:
+
+    @task(task_id="data")
+    def data_task(**context):
+        # (Neptune) It is recommended to pass the api_token and project using
+        #           airflow variables especially when tasks are run on different
+        #           machines.
```
Comment (normandy7) [inline integrations-and-supported-tools/airflow/scripts/Neptune_Airflow.py:98]:
```suggestion
        # (Neptune) We recommend passing the Neptune API token and project name using
        #           Airflow variables, especially when tasks are run on different
        #           machines.
```

**Review by normandy7 — COMMENTED** [review]
Some minor suggestions; looks good otherwise.

**Comment by SiddhantSadangi** [comment]
TODO:

 - [ ] Mark script as failed if DAG fails
 - [ ] Check `NeptuneUnsupportedType` and deprecation warnings

**Review by normandy7 — DISMISSED** [review]

**Review by SiddhantSadangi — APPROVED** [review]

**Pull request merged**